### ISSUE TITLE
mod_rank bug

### ISSUE THREAD

**Issue by sallyrobson** [e4st-dev/E4ST.jl#197]
found some places where we were improperly defining mod_rank which was causing issues in mod order. mod_rank() should dispatch on a type not on a modification itself so the definition should look something like: 
` mod_rank(::Type{<:CustomMod}) ` 

not ` mod_rank(m::CustomMod) `

**Comment by Ethan-Russell** [comment]
We don't have any mods that allow you to set their rank as an input, but we may want to do that eventually.  If we are requiring that to be in the type, we will have to parameterize the type, as in below:

```julia
struct MyCustomMod{R}
    field1
    field2
    function MyCustomMod(field1, field2, rank)
        return new{rank}(field1, field2)
    end
end
```

**Comment by Ethan-Russell** [comment]
This was closed in #198
